src/main.cc

```diff
@@ -677,7 +677,11 @@ mainHandleCommandLineOption(const int optId, const char *optValue)
         if (strlen(SQUID_BUILD_INFO))
             printf("%s\n",SQUID_BUILD_INFO);
 #if USE_OPENSSL
+#if defined(SSLEAY_VERSION)
         printf("\nThis binary uses %s. ", SSLeay_version(SSLEAY_VERSION));
+#else
+        printf("\nThis binary uses %s. ", OpenSSL_version(OPENSSL_VERSION));
```

Comment:
My idea was to just use the new API in the high-level code, and deal with renames in the shim, but @yadij had concerns about licensing changes when using the new names.  That's why I changed it to use the `#if`.  I can't see how the use of new function names can even affect licensing in this case, after reading the simple openssl license file, and particularly since the source can be compiled with a different library anyway. However, I'm not a lawyer.  Please advise me how to deal with this.  If I can have a vote, it would be to use the new API and add use the `#if` in the shim.

Reply:
The TLD;DR answer is yes a shim is fine.

Changing this line does not change the license, quite the reverse. This line is informing about the license conditions imposed from old SSLEAY and OpenSSL versions copyright.

OpenSSL project are on the cusp of changing their license and I was expecting the SSLEAY_VERSION to be removed in versions where SSLEAY no longer applies. OPENSSL_VERSION is expected to remain in all OpenSSL builds regardless of license.

For our part we only need this text where the old SSLEAY legal situation applies to the library. So I was asking about a check primarily to see whether we could use that version macro as indicator of that state - the answer to that was no. Unfortunately for us the OpenSSL versions still currently have the old OpenSSL-specific clauses even when the SSLEAY clauses do not apply.

Reply:
I'm using the new API for these calls.  I meant to have done it already, but I missed it, sorry.